driver-async/src/main/com/mongodb/async/client/gridfs/GridFSFileCodecProvider.java

```diff
@@ -0,0 +1,17 @@
+package com.mongodb.async.client.gridfs;
+
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
+
+public class GridFSFileCodecProvider implements CodecProvider {
```

Comment:
Probably can use org.bson.codecs.configuration.RootCodecRegistry#withCodec instead of needing this class at all.